Sources/PointFree/Subscribe.swift

```diff
@@ -27,94 +28,139 @@ private let validateSubscribeData
     <<< validateCoupon
     <<< validateReferrer
 
-private func subscribe(_ conn: Conn<StatusLineOpen, Tuple3<User, SubscribeData, Referrer?>>)
-  -> IO<Conn<ResponseEnded, Data>> {
-
-    let referralDiscount: Cents<Int> = -18_00
+private func subscribe(
+  _ conn: Conn<StatusLineOpen, Tuple3<User, SubscribeData, Referrer?>>
+) -> IO<Conn<ResponseEnded, Data>> {
+
+  let (user, subscribeData, referrer) = lower(conn.data)
+  let referrerDiscount: Cents<Int> =
+    referrer?.stripeSubscription.discount?.coupon.id == Current.envVars.regionalDiscountCouponId
+      ? -9_00
+      : -18_00
+  let referredDiscount: Cents<Int> = subscribeData.useRegionalDiscount
+    ? -9_00
+    : -18_00
+
+  let stripeSubscription = Current.stripe.createCustomer(
+    subscribeData.token,
+    user.id.rawValue.uuidString,
+    user.email,
+    nil,
+    subscribeData.pricing.interval == .year ? referrer.map(const(referredDiscount)) : nil
+  )
+    .flatMap { customer -> EitherIO<Error, Stripe.Subscription> in
+      let country = customer.sources.data.first?.left?.country
+
+      guard country != nil || !subscribeData.useRegionalDiscount else {
+        return throwE(
+          StripeErrorEnvelope(
+            error: .init(
+              message: """
+                Couldn't verify issue country on credit card. Please try another credit card.
+                """
+            )
+          )
+        )
+      }
 
-    let (user, subscribeData, referrer) = lower(conn.data)
+      guard !subscribeData.useRegionalDiscount
+        || DiscountCountry.all.contains(where: { $0.countryCode == country })
```

Comment:
this guards against using the regional discount when the CC country is not in our list of allowed ones.